.gitlab-ci.yml

```diff
@@ -14,19 +14,65 @@ variables:
   BUILD_DIR: build
 
 stages:
+  - build
   - test
   - test_acceptance_build
   - test_acceptance
-  - build
   - publish
 
+compile:
+  image: golang:1.14
+  stage: build
+  before_script:
+    - mkdir -p /go/src/$(dirname $REPO_NAME)/mender-cli /go/src/_/builds
+    - cp -r $CI_PROJECT_DIR /go/src/$(dirname $REPO_NAME)
+    - cd /go/src/$(dirname $REPO_NAME)/mender-cli
+    - make get-build-deps
+  script:
+    - make build-multiplatform
+    - cp /go/src/$(dirname $REPO_NAME)/mender-cli/mender-cli.linux.amd64 $CI_PROJECT_DIR/
+    - cp /go/src/$(dirname $REPO_NAME)/mender-cli/mender-cli.darwin.amd64 $CI_PROJECT_DIR/
+  artifacts:
+    paths:
+      - mender-cli.linux.amd64
+      - mender-cli.darwin.amd64
+
+test:smoketests:mac:
+  stage: test
+  dependencies:
+    - compile
+  before_script: []
+  script:
+    - ./mender-cli.darwin.amd64
+  tags:
+    - mender-mac-slave
+
+test:smoketests:linux:
+  image: debian:9
+  stage: test
+  dependencies:
+    - compile
+  before_script: []
+  script:
+    - ./mender-cli.linux.amd64
+
+test:static:  
+  image: golang:1.14
+  stage: test
+  script:
+    - make get-go-tools
+    - make test-static
+
 test_acceptance_build:tools:
   image: golang:1.14
   stage: test_acceptance_build
+  dependencies:
+    - compile
```

Comment:
Question:
Should this not be `build`? Or can you have deps on single jobs too?

Reply:
yep, deps on single jobs determine what artifacts will be available in your job